postgres/datadog_checks/postgres/postgres.py

```diff
@@ -423,6 +425,7 @@ def check(self, _):
             self.log.debug("Running check against version %s", str(self.version))
             self._collect_stats(tags)
             self._collect_custom_queries(tags)
+            self.statement_metrics.collect_per_statement_metrics(self, self.db, tags)
```

Comment:
It would also be good to guard against this collection by checking `deep_database_monitoring` here since it's more explicit than having the condition within the collection function itself.